src/ScoreCard.js

```diff
@@ -0,0 +1,81 @@
+function ScoreCard() {
+  this.score = 0;
+  this.currentFrame = 1;
+  this.totalFrames = 10;
+  this.maxFrames = 12;
+  this.currentBowl = 1;
+  this.isGameOver = false;
+  this.frames = [];
+  for (frameIndex=0; frameIndex <= this.maxFrames; frameIndex +=1 ) this.frames[frameIndex] = {
+    bowl1: 0, 
+    bowl2: 0, 
+    score: 0,
+    strike: false,
+    spare: false,
+  }
+};
+
+ScoreCard.prototype.hitPins = function(numberOfPins){
+  if (this.currentBowl === 1) {
+    this.frames[this.currentFrame].bowl1 = numberOfPins;
+    this.currentBowl += 1
+    if ((this.currentFrame === this.totalFrames + 1 & this.frames[this.totalFrames].spare === true) || (this.currentFrame === this.maxFrames)) {
+      this.isGameOver = true;
+    }
+    if (numberOfPins === 10) {
+      this.frames[this.currentFrame].strike = true;
+      this.changeFrame();
+    }
+
+  } else if (this.currentBowl === 2 ) {
+      this.frames[this.currentFrame].bowl2 = numberOfPins;
+      if (this.frames[this.currentFrame].bowl1 + numberOfPins === 10) {
+        this.frames[this.currentFrame].spare = true;
+      }
+      if ((this.currentFrame === this.totalFrames && this.frames[this.currentFrame].spare === false) || (this.currentFrame === this.totalFrames + 1)) {
+        this.isGameOver = true
+      }
+      this.changeFrame();
+    }
+};
```

Comment:
Avoid:

- nested conditionals (these can be decomposed into helper methods)
- unclear control flow: every time you have these stacked if statements, it's not clear if one, none, or all branches fire. A lot of these feel like they could be their own methods